travis/apis.py

```diff
@@ -122,6 +123,19 @@ def _component_lock(self, component, lock=True):
             time.sleep(60)
         return True
 
+    def _component_commit(self, component):
+        url = (self.host + '/components/%s/%s/repository/' %
+               (self.project['slug'], component['slug']))
+        needs_commit = self._request(url)
+        if not needs_commit['needs_commit']:
+            return
+        for i in range(10):
```

Comment:
The loop is only one validation extra, if the api can not make the commit or another trouble related with the API
If all goes well the loop only make one iteration